code/game/gamemodes/malfunction/Malf_Modules.dm

```diff
@@ -59,6 +59,14 @@
 	var/timer = 450
 
 /obj/machinery/doomsday_device/process()
+	if(z != ZLEVEL_STATION)
+		minor_announce("DOOMSDAY DEVICE OUT OF RANGE OF STATION, ABORTING", "ERROR ER0RR $R0RRO$!R41.%%!!(%$^^__+ @#F0E4", 1)
+		SSshuttle.emergencyNoEscape = 0

```

Comment:
You can only buy one device.

The timer reset is intended so people have time to get to the shuttle after killing the AI